### PR TITLE
Modifications for rag

### PR DESCRIPTION
**Pull request by RoniDataloopAi**

### CODE REVIEW

File: models/base_model_adapter.py [hunk 1]
```diff
@@ -41,88 +39,97 @@ def extract_content(line):
...
         }
-        full_answer = self.post_stream(url=self.url, data=data, headers=headers, stream=stream)
 
-        return full_answer
+        s = requests.Session()
+        response = s.post(self.url, data=json.dumps(data), headers=headers, stream=self.stream)
+        if self.stream:
+            try:
+                with response:  # To properly closed The response object after the block of code is executed
+                    if not response.ok:
```
Comment (ShadiMah) [inline models/base_model_adapter.py]: can be moved directly under the request?

File: models/base_model_adapter.py [hunk 2]
```diff
@@ -41,88 +39,97 @@ def extract_content(line):
...
+                        raise ValueError(f'error:{response.status_code}, message: {response.text}')
+                    logger.info("Streaming the response")
+                    for line in response.iter_lines():
+                        if line:
+                            line = line.decode('utf-8')
+                            yield self.extract_content(line) or ""
+            except requests.exceptions.RequestException as e:
+                logger.warning(f"Request failed: {e}")
```
Comment (ShadiMah) [inline models/base_model_adapter.py]: if request failed shouldn't it raise an error?

**Review by ShadiMah — CHANGES_REQUESTED** [review]

File: models/base_model_adapter.py (hunk 1 above)
Comment (RoniDataloopAi) [reply models/base_model_adapter.py]: i'll move the response.ok check before, stream has to be a condition to read the response

File: models/base_model_adapter.py (hunk 2 above)
Comment (RoniDataloopAi) [reply models/base_model_adapter.py]: yes, i'll change it

**Review by ShadiMah — APPROVED** [review]

**Pull request merged**